code/modules/mob/mob_helpers.dm

```diff
@@ -111,6 +111,20 @@
 		return 1
 	return 0
 
+/proc/iszombie(A)
```

Comment:
Adjust `isundead()` too. Just make it call this proc itself instead of doing all its own separate zombie checks.